packages/sdks/web-component/src/lib/descope-wc/sdkScripts/grecaptcha.ts

```diff
@@ -84,6 +92,46 @@ export const loadGRecaptcha = (
     getNewToken(grecaptchaInstance, elementRef);
   };
 
+  /**
+   * Checks if the reCAPTCHA token needs refreshing and refreshes it if necessary
+   * This is called before form submission to ensure we have a valid token
+   * @returns Promise that resolves when token is refreshed or if refresh isn't needed
+   */
+  const refreshIfTokenExpired = async (): Promise<void> => {
+    const currentTime = Date.now();
+    const timeDiff = currentTime - lastTokenFetchTime;
+
+    if (timeDiff > TOKEN_REFRESH_TIME) {
+      stopTimer();
+      const prev = lastTokenFetchTime;
+      resumeScriptExecution();
+
+      // Return a promise that resolves once the token is refreshed or times out
+      return new Promise<void>((resolve) => {
+        // Set a timeout to prevent indefinite waiting
+        const timeout = setTimeout(() => {
+          // eslint-disable-next-line no-console
+          console.warn('reCAPTCHA token refresh timed out');
+          resolve(); // Resolve anyway to prevent blocking form submission
+        }, 5000); // 5 second timeout for token refresh
+
+        const checkToken = () => {
+          if (lastTokenFetchTime !== prev) {
+            clearTimeout(timeout);
+            resolve();
+          } else {
+            setTimeout(checkToken, 150);
+          }
```

Comment:
👍  but why 150? why not 1000 or more? tickets should be in seconds probably

Reply:
If it will be 1000, so it will always take at least 1000 sec of waiting